### ISSUE TITLE
Metadata enhancements

### ISSUE THREAD

**Issue by thj01** [brrd/Abricotine#222]
Is it possible to make a metadata section, which could translate into placeholders eg:

author
email
date
web (authors web address)
license
keywords (tags)
wordcount
charactercount

example

::author: Thomas __produces__  $author = Thomas

That would make it possible with HTML and CSS to insert this, and design advanced webcontent both for web wieving and PDF printing, that could be used in teaching context.

It would be awesome, if it could be done with a _.json_ file so it was possible to make own extensions

example
``` json
author {
"metadata_string":"author",
"placeholder":"$author"
}

**Comment by brrd** [comment]
Is there any standardized way to do this in markdown? Maybe YAML headers (see #142)? 
I don't want to reinvent the wheel and Abricotine doesn't aim to create new standards.

**Comment by thj01** [comment]
As i read it the YAML headers could do the trick ... I'm no programmer ... only at user/teacher 

thx for the reply

**Comment by dskopa-softdev** [comment]
Using YAML frontmatter blocks for meta information right inside markdown files seems to be a common approach for this.

```
---
title: An Article
summary: A story about metadata
author: Some Author
creation_date: 2018-05-17
tags:
- metadata
- Abricotine
---

# What is this
Here goes the content of markdown file
```

In view mode this block can be hidden or displayed as a small grayed out text. When editing consider it's just a text block (for now)

I know Github, Hugo (md -> html site generator) and some other soft support frontmatter for storing metadata in markdown

**Comment by seifferth** [comment]
Actually, I believe that this issue is mostly resolved with YAML-support and the pandoc task (also see pull request #271 for generating complete documents with pandoc). Once #271 is merged, you should be able to simply save html files (with metadata included) using the pandoc task. It is also possible to save Word-Documents (docx) and many other formats by simply specifying the desired file extension. See also <https://pandoc.org/MANUAL.html#option--to>.

**Comment by brrd** [comment]
I merged #271. YAML header will be supported in next release by using pandoc (see @seifferth comment).
Thanks.
